frontend/src/components/worksheets/items/SchemaItem/SchemaItem.js

```diff
@@ -0,0 +1,405 @@
+        let curRow = [...this.state.rows];
+        curRow.splice(idx + 1, 0, newRow);
+        this.setState({ rows: curRow, newAddedRow: idx + 1 });
+    };
+
+    changeFieldValue = (idx, key) => (e) => {
+        if (!this.props.editPermission) return;
+        const { rows } = this.state;
+        let copiedRows = [...rows];
+        copiedRows.splice(idx, 1, { ...rows[idx], [key]: e.target.value });
+        this.setState({ rows: [...copiedRows] });
+    };
+
+    changeSchemaName = (e) => {
+        if (!this.props.editPermission) return;
+        this.setState({ cur_schema_name: e.target.value });
+    };
+
+    moveFieldRow = (idx, direction) => () => {
+        if (!this.props.editPermission) return;
+        // -1 for moving up, 1 for moving down
+        const { rows } = this.state;
+        let copiedRows = [...rows];
+        let newIndex = idx + direction;
+        [copiedRows[newIndex], copiedRows[idx]] = [copiedRows[idx], copiedRows[newIndex]];
+        if (copiedRows[idx].from_schema_name !== this.props.item.schema_name) {
+            // if the last row we switched with was generated by addschema
+            // we should check and keep switching
+            // until we meet a non-addschema row or top/end of table
+            idx += direction;
+            newIndex += direction;
+            while (
+                newIndex >= 0 &&
+                newIndex < rows.length &&
+                rows[newIndex].from_schema_name !== this.props.item.schema_name
+            ) {
+                [copiedRows[newIndex], copiedRows[idx]] = [copiedRows[idx], copiedRows[newIndex]];
+                idx += direction;
+                newIndex += direction;
+            }
+        }
+        this.setState({ rows: copiedRows });
+    };
+
+    removeFieldRow = (idx) => () => {
+        if (!this.props.editPermission) return;
+        const { rows } = this.state;
+        let copiedRows = [...rows];
+        copiedRows.splice(idx, 1);
+        this.setState({ rows: copiedRows });
+    };
+
+    componentDidUpdate(prevProps, prevState) {
+        if (
+            prevProps.item.field_rows !== this.props.item.field_rows ||
+            prevProps.item.schema_name !== this.props.item.schema_name
+        ) {
+            this.setState({
+                rows: [...this.props.item.field_rows],
+                cur_schema_name: this.props.item.schema_name,
+            });
+        }
+        if (this.state.newAddedRow !== -1 && this.state.rows.length === prevState.rows.length + 1) {
+            document.getElementById('textbox-' + this.state.newAddedRow + '-0').focus();
+        }
+    }
+
+    render() {
+        const { classes, editPermission, focused, item } = this.props;
+        const { showSchemaDetail, rows } = this.state;
+        const schemaItem = item;
+        const schemaHeaders = schemaItem.header;
+        const schema_name = schemaItem.schema_name;
+        let headerHtml, bodyRowsHtml;
+        const explanations = {
```

Comment:
In general, "field", "generalized-path", "post-processor" aren't very intuitive, but that's the official names according to the code and documentation.  Maybe instead of having the gray placeholder text be '<none>', use 'display this column name', 'retrieve value from this path', 'how to render value', respectively.